kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/SinkContextReader.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017-2023 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.aws.s3.sink
+
+import cats.implicits.catsSyntaxEitherId
+
+import java.util
+import java.util.Collections
+import scala.jdk.CollectionConverters.MapHasAsJava
+import scala.jdk.CollectionConverters.MapHasAsScala
+import scala.util.Try
+
+object SinkContextReader {
+
+  def mergeProps(
+    contextPropsFn: () => util.Map[String, String],
+  )(fallbackProps:  util.Map[String, String],
+  ): Either[String, util.Map[String, String]] = {
+    val context  = Try(contextPropsFn()).toOption.getOrElse(Collections.emptyMap())
```

Comment:
Please help me understand the context a bit. Maybe adding a comment will help further. 
When would a config builder (contextPropsFn) fail? 

Why would the fallbackProps be null?
We always return success so is there any value having Either[String, util.Map...]?

Reply:
I think this was in response to your comment:

https://github.com/lensesio/stream-reactor/pull/937#discussion_r1184726811

Reply:
> When would a config builder (contextPropsFn) fail?

1. Internal error in Kafka connect where the task id is not recognised.  Very unlikely.
2. When we write integration tests but do not want to mock the context every time. (real answer)

> Why would the fallbackProps be null?
I don't think they would.

> We always return success so is there any value having Either[String, util.Map...]?
Good point.  Consider it gone.